### PR TITLE
fix for awssdk 1.9.16

### PR DESCRIPTION
**Pull request by mirceal**
suggested fix to make the txlib compatible with newer awssdk

### CODE REVIEW

File: src/main/java/com/amazonaws/services/dynamodbv2/transactions/Request.java [hunk 1]
```diff
@@ -444,11 +451,33 @@ protected static Request deserialize(String txId, ByteBuffer rawRequest) {
...
+        @JsonIgnore
+        public abstract void setProgressListener(ProgressListener progressListener);
+
     }
-    
+  
+    private static abstract class ExpectedAttributeValueMixIn {
+      
+      @JsonIgnore
+      public abstract void setComparisonOperator(String comparisonOperator);
```
Comment (johnnyw-aws) [inline src/main/java/com/amazonaws/services/dynamodbv2/transactions/Request.java]: might also need to JsonIgnore
setComparisonOperator(ComparisonOperator comparisonOperator)

**Comment by johnnyw-aws** [comment]
I am slightly hesitant to pull it directly into master because there may need to be a few more updates to add validation to reject requests that are using the expressions language (similar to how existing conditional put requests are rejected in the doValidate... code in Request.java)

Would a separate branch be fine? Or if you can add the validation, I think it should be ok to merge into master.

**Comment by mirceal** [comment]
the thinking behind this is to get it running on the latest (or somewhat latest) version of the SDK
I hacked the absolute minimum for this fix in an hour or two of time I had available to get it working - I'm not intimately familiar with the inner workings of the SDK. 
I though to give this back and to be able to leverage this library without worrying about maintaining my own fork.

Someone who wrote the TxLib taking a look would make me feel much better.

**Comment by johnnyw-aws** [comment]
Ok, I made a separate branch "dev/1.1.9"

Can you make a pull request for these changes into there instead? Then we can make sure all the necessary changes are present before merging it into mainline.

**Comment by mirceal** [comment]
created https://github.com/awslabs/dynamodb-transactions/pull/19

**Pull request closed without merging**